x-pack/plugins/index_management/__jest__/client_integration/home/home.test.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { act } from 'react-dom/test-utils';
+
+import { setupEnvironment, nextTick } from '../helpers';
+
+import { HomeTestBed, setup } from './home.helpers';
+
+describe('<IndexManagementHome />', () => {
+  const { server, httpRequestsMockHelpers } = setupEnvironment();
+  let testBed: HomeTestBed;
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  describe('on component mount', () => {
+    beforeEach(async () => {
+      httpRequestsMockHelpers.setLoadIndicesResponse([]);
+
+      testBed = await setup();
+
+      await act(async () => {
+        const { component } = testBed;
+
+        await nextTick();
+        component.update();
+      });
+    });
+
+    test('should set the correct app title', () => {
+      const { exists, find } = testBed;
+      expect(exists('appTitle')).toBe(true);
+      expect(find('appTitle').text()).toEqual('Index Management');
+    });
+
+    test('should have a link to the documentation', () => {
+      const { exists, find } = testBed;
+      expect(exists('documentationLink')).toBe(true);
+      expect(find('documentationLink').text()).toBe('Index Management docs');
+    });
+
+    describe('tabs', () => {
+      test('should have 2 tabs', () => {
+        const { find } = testBed;
+        const templatesTab = find('templatesTab');
+        const indicesTab = find('indicesTab');
+
+        expect(indicesTab.length).toBe(1);
+        expect(indicesTab.text()).toEqual('Indices');
+        expect(templatesTab.length).toBe(1);
+        expect(templatesTab.text()).toEqual('Index Templates');
+      });
+
+      test('should navigate to Index Templates tab', async () => {
+        const { exists, actions, component } = testBed;
+
+        expect(exists('indicesList')).toBe(true);
+        expect(exists('templateList')).toBe(false);
+
+        httpRequestsMockHelpers.setLoadTemplatesResponse([]);
+
+        actions.selectHomeTab('templatesTab');
+
+        await act(async () => {
```

Comment:
info: This block is exactly what we want to remove everywhere. It creates latency in the tests and could be the cause of flakiness.

It should be written

```js
act(() => {
  // this is causing states updates. There can only be **one** in each act() call
  // It should be synchronous if possible.
  // Sometimes hook update require async, so if there is an error in console --> use async act()
  actions.selectHomeTab('templatesTab');
});

// If I need a second action straightaway, we use another act()
act(() => {
  actions.doSomethingElse();
});

// And we only need to call update() once after all actions
component.update();
```

Reply:
As an alternative, we could wrap the `act()` inside our helpers' actions and keep the test file cleaner.